R/validate.R

```diff
@@ -80,6 +80,28 @@ validate_ww_conc_data <- function(ww_data,
   assert_non_missingness(site_pops, arg, call)
   assert_elements_non_neg(site_pops, arg, call)
 
+  # Check that there are no repeated site populations
+  records_per_site_per_pop <- ww_data |>
+    dplyr::select("site", "site_pop") |>
+    unique() |>
+    dplyr::group_by(.data$site) |>
+    dplyr::summarize(n = dplyr::n())
+
+  if (any(records_per_site_per_pop$n != 1)) {
+    stop(
+      "Package expects that there is only one site population per site.",
+      " Got site(s) with more than one population: ",
+      paste0(
+        records_per_site_per_pop$site[records_per_site_per_pop$n > 1],
+        " (",
+        records_per_site_per_pop$n[records_per_site_per_pop$n > 1],
+        " records)",
+        collapse = ", "
+      )
+    )
+  }
```

Comment:
This is the check I'm implementing, @kaitejohnson. LMK if this makes sense to you. The CI is not passing.

Reply:
Cool looks like the test of preprocess ww data is failing -- I will investigate!

Reply:
Yep - its failing bc I actually didn;t change the order of the site pops when I changed the order of the sites to be scrambled (basically it failed for the right reason, and hadn't been caught before)

Reply:
Just adjusted (hope its ok that I pushed to your branch directly...)